File: user-service/src/main/java/rs/raf/user_service/client/BankClientFallbackFactory.java
package rs.raf.user_service.client;

import org.springframework.cloud.openfeign.FallbackFactory;
import org.springframework.stereotype.Component;

@Component
public class BankClientFallbackFactory implements FallbackFactory<BankClient> {
    @Override
    public BankClient create(Throwable cause) {
        return new BankClient() {
            @Override
            public void confirmPayment(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void confirmTransfer(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void changeAccountLimit(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void approveCardRequest(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void rejectConfirmPayment(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void rejectConfirmTransfer(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void rejectChangeAccountLimit(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }

            @Override
            public void rejectApproveCardRequest(Long id) {
                throw new RuntimeException("Unable to communicate with Bank Service");
            }
        };
    }
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/external/BankClientFallbackFactory.kt
package pack.userservicekotlin.external

import org.springframework.cloud.openfeign.FallbackFactory
import org.springframework.stereotype.Component

@Component
class BankClientFallbackFactory : FallbackFactory<BankClient> {
    override fun create(cause: Throwable): BankClient =
        object : BankClient {
            override fun confirmPayment(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun confirmTransfer(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun changeAccountLimit(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun approveCardRequest(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun rejectConfirmPayment(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun rejectConfirmTransfer(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun rejectChangeAccountLimit(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")

            override fun rejectApproveCardRequest(id: Long?): Unit = throw RuntimeException("Unable to communicate with Bank Service")
        }
}
